### PR TITLE
fix: remove initializer decorator from footer links

### LINKED ISSUE #974: [bug] footer: TypeError Cannot read properties of undefined (reading 'warn')
### Describe the bug

Working with the @rhds-elements packaged using especifically the rh-footer, our sentry is catching this error:

image

It seems that the logger instance is not found to log the warning.

We notice that the error happens with a specific version of the google chrome:

Screenshot from 2023-05-03 21-53-05

We were not able to reproduce this error locally but it is happening with quite frequency in production. It's not breaking our application but it would be nice to discover the problem.

### Which Component?

rh-footer

### Steps to reproduce

1. go to this url: https://skills.ole.redhat.com/en
2. in production the error is being dispatched just doing this, also you might need to use the google chrome version: 112.0.5615

### Expected behaviour

When access this https://skills.ole.redhat.com/en the error should not be dispatched.

### Screenshots

_No response_

### Element HTML

```HTML
return (<rh-footer data-analytics-region="page-footer"></rh-footer>)}
```

### Operating System (OS)

Linux

### Browser

Chrome

### More browser information

- chrome: 112.0.5615

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by bennypowers**
fixes #974

## What I did

1. remove the `@initializer` decorator from rh-footer-links and re-implement it's functionality with a `MutationObserver`
2. minor refactors (real `#private` fields)

## Testing Instructions

1. This is a hard one to test, but it should address the issue on chrome 112 described at #974. at the very least it shouldn't break anything

You might try testing like so:

Check out this branch, then

```sh
npm run build
npm link
```

```sh
cd /tmp
npm init vite footer-links-react-test # choose "React, javascript"
cd footer-links-react-test
npm link @rhds/elements
```

Replace the contents of `src/App.jsx` with the following:

<details><summary>snippet</summary>

```jsx
import '@rhds/elements/rh-footer/rh-footer.js';
import '@rhds/elements/rh-footer/rh-footer-lightdom.css';

function App() {
  return (
    <>
    <rh-footer data-analytics-region="page-footer"
               color-palette="darker">
      <a slot="logo" style={{width: '180px'}} href="https://redhat.com/en" data-analytics-category="Footer" data-analytics-text="Logo">
        <img alt="Red Hat logo" src="https://static.redhat.com/libs/redhat/brand-assets/2/corp/logo--on-dark.svg" loading="lazy"/>
      </a>
      <rh-footer-social-link slot="social-links" icon="web-icon-linkedin">
        <a href="https://www.linkedin.com/company/red-hat"><pfe-icon icon="web-icon-linkedin" pfe-size="md" size="md" aria-hidden="true" pfelement="" class="PFElement" type="content"></pfe-icon></a>
      </rh-footer-social-link>
      <rh-footer-social-link slot="social-links" icon="web-icon-youtube">
        <a href="https://www.youtube.com/user/RedHatVideos"><pfe-icon icon="web-icon-youtube" pfe-size="md" size="md" aria-hidden="true" pfelement="" class="PFElement" type="content"></pfe-icon></a>
      </rh-footer-social-link>
      <rh-footer-social-link slot="social-links" icon="web-icon-facebook">
        <a href="https://www.facebook.com/redhatinc"><pfe-icon icon="web-icon-facebook" pfe-size="md" size="md" aria-hidden="true" pfelement="" class="PFElement" type="content"></pfe-icon></a>
      </rh-footer-social-link>
      <rh-footer-social-link slot="social-links" icon="web-icon-twitter">
        <a href="https://twitter.com/RedHat"><pfe-icon icon="web-icon-twitter" pfe-size="md" size="md" aria-hidden="true" pfelement="" class="PFElement" type="content"></pfe-icon></a>
      </rh-footer-social-link>
      <h3 slot="links-0" data-analytics-text="Products" id="rh-footer-b3nfw6erk">Products</h3>
      <ul slot="links-1" aria-labelledby="rh-footer-b3nfw6erk">
        <li><a href="https://redhat.com/en/technologies/linux-platforms/enterprise-linux" data-analytics-category="Footer|Products" data-analytics-text="Red Hat Enterprise Linux">Red Hat Enterprise Linux</a></li>
        <li><a href="https://redhat.com/en/technologies/cloud-computing/openshift" data-analytics-category="Footer|Products" data-analytics-text="Red Hat OpenShift">Red Hat OpenShift</a></li>
        <li><a href="https://redhat.com/en/technologies/management/ansible" data-analytics-category="Footer|Products" data-analytics-text="Red Hat Ansible Automation Platform">Red Hat Ansible Automation Platform</a></li>
        <li><a href="https://redhat.com/en/technologies/cloud-computing/openshift/cloud-services" data-analytics-category="Footer|Products" data-analytics-text="Cloud services">Cloud services</a></li>
        <li><a href="https://redhat.com/en/technologies/all-products" data-analytics-category="Footer|Products" data-analytics-text="See all products">See all products</a></li>
      </ul>
      <h3 slot="links-2" data-analytics-text="Tools" id="rh-footer-3rqyacjjq">Tools</h3>
      <ul slot="links-3" aria-labelledby="rh-footer-3rqyacjjq">
        <li><a href="https://sso.redhat.com" data-analytics-category="Footer|Tools" data-analytics-text="My account">My account</a></li>
        <li><a href="https://redhat.com/en/services/training-and-certification" data-analytics-category="Footer|Tools" data-analytics-text="Training and certification">Training and certification</a></li>
        <li><a href="https://access.redhat.com" data-analytics-category="Footer|Tools" data-analytics-text="Customer support">Customer support</a></li>
        <li><a href="https://developers.redhat.com/" data-analytics-category="Footer|Tools" data-analytics-text="Developer resources">Developer resources</a></li>
        <li><a href="https://learn.redhat.com/" data-analytics-category="Footer|Tools" data-analytics-text="Learning community">Learning community</a></li>
        <li><a href="https://connect.redhat.com/" data-analytics-category="Footer|Tools" data-analytics-text="Partner resources">Partner resources</a></li>
        <li><a href="https://redhat.com/en/resources" data-analytics-category="Footer|Tools" data-analytics-text="Resource library">Resource library</a></li>
      </ul>
      <h3 slot="links-4" data-analytics-text="Try buy sell" id="rh-footer-83w1yt6g8">Try, buy &amp; sell</h3>
      <ul slot="links-5" aria-labelledby="rh-footer-83w1yt6g8">
        <li><a href="https://redhat.com/en/products/trials" data-analytics-category="Footer|Try buy sell" data-analytics-text="Product trial center">Product trial center</a></li>
        <li><a href="https://marketplace.redhat.com" data-analytics-category="Footer|Try buy sell" data-analytics-text="Red Hat Marketplace">Red Hat Marketplace</a></li>
        <li><a href="https://catalog.redhat.com/" data-analytics-category="Footer|Tools" data-analytics-text="Red Hat Ecosystem Catalog">Red Hat Ecosystem Catalog</a></li>
        <li><a href="http://redhat.force.com/finder/" data-analytics-category="Footer|Try buy sell" data-analytics-text="Find a partner">Find a partner</a></li>
        <li><a href="https://www.redhat.com/en/store" data-analytics-category="Footer|Try buy sell" data-analytics-text="Red Hat Store">Red Hat Store</a></li>
        <li><a href="https://cloud.redhat.com/" data-analytics-category="Footer|Tools" data-analytics-text="Console">Console</a></li>
      </ul>
      <h3 slot="links-6" data-analytics-text="Communicate" id="rh-footer-amqxtn903">Communicate</h3>
      <ul slot="links-7" aria-labelledby="rh-footer-amqxtn903">
        <li><a href="https://redhat.com/en/services/consulting-overview#contact-us" data-analytics-category="Footer|Communicate" data-analytics-text="Contact consulting">Contact consulting</a></li>
        <li><a href="https://redhat.com/en/contact" data-analytics-category="Footer|Communicate" data-analytics-text="Contact sales">Contact sales</a></li>
        <li><a href="https://redhat.com/en/services/training-and-certification/contact-us" data-analytics-category="Footer|Communicate" data-analytics-text="Contact training">Contact training</a></li>
        <li><a href="https://redhat.com/en/about/social" data-analytics-category="Footer|Communicate" data-analytics-text="Social">Social</a></li>
      </ul>
      <rh-footer-block slot="main-secondary">
        <h3 slot="header" data-analytics-text="About Red Hat">About Red Hat</h3>
        <p> We’re the world’s leading provider of enterprise open source solutions—including Linux, cloud, container, and Kubernetes. We deliver hardened solutions that make it easier for enterprises to work across platforms and environments, from the core datacenter to the network edge.</p>
      </rh-footer-block>
      <rh-footer-block slot="main-secondary">
        <h3 slot="header" data-analytics-text="Subscribe to our newsletter Red Hat Shares">Subscribe to our newsletter, Red Hat Shares</h3>
        <rh-cta><a href="https://redhat.com/en/email-preferences?newsletter=RH-Shares&amp;intcmp=7016000000154xCAAQ" data-analytics-category="Footer|About Red Hat" data-analytics-text="Sign up now">Sign up now</a></rh-cta>
      </rh-footer-block>
      <rh-global-footer slot="global" color-palette="darker">
        <h3 slot="links-primary" data-analytics-text="Red Hat legal and privacy links" id="rh-footer-kuk0z3mjk" hidden="">Red Hat legal and privacy links</h3>
        <ul slot="links-primary" data-analytics-region="page-footer-bottom-primary" aria-labelledby="rh-footer-kuk0z3mjk"><li><a href="https://redhat.com/en/about/company" data-analytics-category="Footer|Corporate" data-analytics-text="About Red Hat">About Red Hat</a></li><li><a href="https://redhat.com/en/jobs" data-analytics-category="Footer|Corporate" data-analytics-text="Jobs">Jobs</a></li><li><a href="https://redhat.com/en/events" data-analytics-category="Footer|Corporate" data-analytics-text="Events">Events</a></li><li><a href="https://redhat.com/en/about/office-locations" data-analytics-category="Footer|Corporate" data-analytics-text="Locations">Locations</a></li><li><a href="https://redhat.com/en/contact" data-analytics-category="Footer|Corporate" data-analytics-text="Contact Red Hat">Contact Red Hat</a></li><li><a href="https://redhat.com/en/blog" data-analytics-category="Footer|Corporate" data-analytics-text="Red Hat Blog">Red Hat Blog</a></li><li><a href="https://redhat.com/en/about/our-culture/diversity-equity-inclusion/statement" data-analytics-category="Footer|Corporate" data-analytics-text="Diversity equity and inclusion">Diversity, equity, and inclusion</a></li><li><a href="https://coolstuff.redhat.com/" data-analytics-category="Footer|Corporate" data-analytics-text="Cool Stuff Store">Cool Stuff Store</a></li><li><a href="https://www.redhat.com/en/summit" data-analytics-category="Footer|Corporate" data-analytics-text="Red Hat Summit">Red Hat Summit</a></li></ul>
        <rh-footer-copyright slot="links-secondary">© 2022 Red Hat, Inc.</rh-footer-copyright>
        <h3 slot="links-secondary" data-analytics-text="Red Hat legal and privacy links" id="rh-footer-17rs2bjg7" hidden="">Red Hat legal and privacy links</h3>
        <ul slot="links-secondary" data-analytics-region="page-footer-bottom-secondary" aria-labelledby="rh-footer-17rs2bjg7"><li><a href="https://redhat.com/en/about/privacy-policy" data-analytics-category="Footer|Red Hat legal and privacy links" data-analytics-text="Privacy statement">Privacy statement</a></li><li><a href="https://redhat.com/en/about/terms-use" data-analytics-category="Footer|Red Hat legal and privacy links" data-analytics-text="Terms of use">Terms of use</a></li><li><a href="https://redhat.com/en/about/all-policies-guidelines" data-analytics-category="Footer|Red Hat legal and privacy links" data-analytics-text="All policies and guidelines">All policies and guidelines</a></li><li><a href="https://redhat.com/en/about/digital-accessibility" data-analytics-category="Footer|Red Hat legal and privacy links" data-analytics-text="Digital accessibility" className="active">Digital accessibility</a></li></ul>
      </rh-global-footer>
    </rh-footer>
    </>
  )
}

export default App

```

</details>

then 
```
npm run dev
```

You should not see the error described in the issue, namely:
> TypeError Cannot read properties of undefined (reading 'warn')

I tried running chromium linux 112.0.5605.0 from this page https://vikyd.github.io/download-chromium-history-version/#/ but could not reproduce. I could not find the exact version line 112.0.5615 for any of the architectures listed there :shrug:

### CODE REVIEW

**Comment by bennypowers** [comment]
ping @heyMP

File: elements/rh-footer/rh-footer-links.ts [hunk 1]
```diff
@@ -13,19 +12,28 @@ export class RhFooterLinks extends LitElement {
...
+  #mo = new MutationObserver(() => this.updateAccessibility());
 
-  @initializer()
-  public updateAccessibility() {
+  protected slots = new SlotController(this, 'header');
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.updateAccessibility();
+    this.#mo.observe(this, { childList: true });
```
Comment (heyMP) [inline elements/rh-footer/rh-footer-links.ts:33]: Just verifying; do we need to observe the childList tree?

**Comment by heyMP** [comment]
I couldn't replicate this using Chrome 133 on Browserstack.  But everything seems to be working the same so I would be fine with the changes.

File: elements/rh-footer/rh-footer-links.ts (hunk 1 above)
Comment (bennypowers) [reply elements/rh-footer/rh-footer-links.ts:33]: this forwards the functionality from the `@initializer` decorator

https://github.com/patternfly/patternfly-elements/blob/898b61d8394ed39f2bb27f1e662a31726ccfa169/core/pfe-core/controllers/light-dom-controller.ts#L36-L45

**Review by heyMP — APPROVED** [review]
Links Got Tremendous Mutations

**Pull request merged**